### PR TITLE
Added langchain-aws package.

### PR DESCRIPTION
**Pull request by 3coins**
## Description
Adds [langchain-aws](https://github.com/langchain-ai/langchain-aws) package.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by andrii-i** [comment]
I built sagemaker-distribution based on this PR with aws-langchain included and a newly-added test cases with `python -m pytest -n auto -vv -rs -k "langchain-aws" --local-image-version 1.8.0`, they pass:

Screenshot 2024-05-09 at 6 26 33 PM

Note irregularities with `python src/main.py create-minor-version-artifacts --base-patch-version=$BASE_PATCH_VERSION --force` command that is supposed to be run after adding `{cpu/gpu}.additional_packages_env.in` into new version folder [accordingly to the docs](https://github.com/aws/sagemaker-distribution/blob/main/CONTRIBUTING.md#for-adding-new-conda-packages-to-sagemaker-distribution). It checks if new version folder exists and if it does, checks for existence of `{cpu/gpu}.additional_packages_env.in ` file and pulls new packages into `{cpu/gpu}.env.in`. But for me it sometimes works incorrectly and overwrites new version folder without checking for `{cpu/gpu}.additional_packages_env.in`  or pulling packages from it.

**Comment by andrii-i** [comment]
Note that if all tests are ran, `langchain-aws` tests pass but tests for 5 other packages fail. For the sake of experiment I tried building 1.8.0 without adding `langchain-aws` or anything else and I’m getting the same errors. 

> FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[keras.test.Dockerfile-required_packages0] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[scipy.test.Dockerfile-required_packages4] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[pandas.test.Dockerfile-required_packages7] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[sm-python-sdk.test.Dockerfile-required_packages8] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[pytorch.examples.Dockerfile-required_packages9] - assert 1 == 0
===================================================================================== 5 failed, 20 passed, 2 skipped, 4 warnings in 7016.70s (1:56:56) =====================================================================================
Traceback (most recent call last):
 File “/local/home/aieroshe/Docs/sagemaker-distribution/src/main.py”, line 432, in <module>
  parse_args(get_arg_parser())
 File “/local/home/aieroshe/Docs/sagemaker-distribution/src/main.py”, line 428, in parse_args
  args.func(args)
 File “/local/home/aieroshe/Docs/sagemaker-distribution/src/main.py”, line 193, in build_images
  _test_local_images(image_ids, args.target_patch_version)
 File “/local/home/aieroshe/Docs/sagemaker-distribution/src/main.py”, line 219, in _test_local_images
  assert exit_code == 0, f”Tests failed with exit codes: {exit_codes} against: {image_ids}”
      ^^^^^^^^^^^^^^
AssertionError: Tests failed with exit codes: [] against: []

All packages for tests of which errors are thrown except `sm-python-sdk` are listed in `v1.8.0/gpu.env.out`. At the same time sm-python-sdk is also not listed in `v1.7.1/gpu.env.out` .

Screenshot 2024-05-09 at 4 30 33 AM

File: test/test_artifacts/v1/langchain-aws.test.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ARG SAGEMAKER_DISTRIBUTION_IMAGE
+FROM $SAGEMAKER_DISTRIBUTION_IMAGE
+
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+CMD ["python", "-c", "import langchain_aws"]
```
Comment (balajisankar15) [inline test/test_artifacts/v1/langchain-aws.test.Dockerfile:6]: Can we add more tests

Comment (andrii-i) [reply test/test_artifacts/v1/langchain-aws.test.Dockerfile:6]: Thank you for giving this PR a look @balajisankar15. Is this blocking / could we add more tests in the follow-up PR?

Comment (3coins) [reply test/test_artifacts/v1/langchain-aws.test.Dockerfile:6]: @balajisankar15 
We can add more tests around importing other sub-modules to test their importability. Anything other than that requires an elaborate setup with boto3 credentials and service endpoint infrastructure (bedrock, kendra, vectorstores etc.) to test each of these components, which we don't plan to do for this library. 
We will update the PR with these and open for a review. Thanks.

**Comment by 3coins** [comment]
@balajisankar15 
Added more import tests to the test file. Can you take a look. Thanks.

**Comment by andrii-i** [comment]
@balajisankar15 @claytonparnell 
Rebased this PR on main and updated the path of `{cpu/gpu}.additional_packages_env.in` to `build_artifacts/v1/v1.9/v1.9.0/`. I also built an image and ran `langchain-aws` tests, both successfully. Please give this another look.

**Review by claytonparnell — APPROVED** [review]

**Comment by 3coins** [comment]
@claytonparnell @balajisankar15 
Is there a plan to merge this PR, is there an ETA that you can provide?

**Review by balajisankar15 — APPROVED** [review]

**Pull request merged**